packages/web-app/src/components/appli/Dashboard/index.jsx

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { Box, Typography } from '@material-ui/core';
+import Skeleton from '@material-ui/lab/Skeleton';
+import { withStyles } from '@material-ui/core/styles';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import AssessmentIcon from '@material-ui/icons/Assessment';
+import { fetchStatisticsCountry } from '../../../actions/Country/GetStatisticsCountry';
+import { fetchStatisticsMassif } from '../../../actions/Massif/GetStatisticsMassif';
+import SpecificsCaves from './components/SpecificsCaves';
+import CavesData from './components/CavesData/index';
+import CavesStatistics from './components/CavesStatistics';
+import Alert from '../../common/Alert';
+
+const Title = withStyles(() => ({
+  root: {
+    paddingLeft: '15px',
+    verticalAlign: 'middle',
+    display: 'inline-flex'
+  }
+}))(Typography);
+
+const TitleBox = withStyles(
+  theme => ({
+    root: {
+      backgroundColor: theme.palette.primary.light,
+      color: '#FFFFFF',
+      padding: '15px 30px',
+      marginBottom: '20px'
+    }
+  }),
+  { withTheme: true }
+)(Box);
+
+const DataBox = withStyles(() => ({
+  root: {
+    margin: '0px 2%',
+    marginTop: '30px'
+  }
+}))(Box);
+
+const DashboardBox = withStyles(
+  theme => ({
+    root: {
+      backgroundColor: theme.palette.backgroundColor,
+      margin: '10px 5%'
+    }
+  }),
+  { withTheme: true }
+)(Box);
+
+const Dashboard = ({ countryId, massifId }) => {
```

Comment:
Suggestion: Avoir possible impossible state and simplify the logic in the component:


We can have a separated prop `type` to tell the component if it either handle a `massif` or a `country`

```jsx
type: PropTypes.oneOf(['massif', 'country']),
```

```suggestion
const Dashboard = ({ id, type }) => {
```


Then we can remove some logic, like 

```jsx
const [isCountry, setIsCountry] = useState();
```

Reply:
We keeped the two ids because `countryId` is a string and `massifId` is a number. If we want to factorize the code as suggested, we will need to put the variable type with a PropTypes as any, which is not a very good idea from our point of view. We can indeed add the `type` variable but still keep the `countryId` and `massifId` as 2 different variables. 
What do you think is better ?

Reply:
Keep as it is right now but add a comment to explain why it is like that (different type for both IDs make it not factorizable)